R/add_resource.R

```diff
@@ -59,6 +71,28 @@ add_resource <- function(package, resource_name, df, schema = NULL) {
     )
   )
 
+  # Check data (df or path)
+  assertthat::assert_that(
+    is.data.frame(data) | is.character(data),
+    msg = "`data` must be a data frame or path(s) to CSV file(s)."
+  )
+  if (is.data.frame(data)) {
+    df <- data
+  } else {
+    # Check existence of files (no further checks)
+    paths <- purrr::map_chr(
+      data, ~ check_path(.x, directory = NULL, unsafe = TRUE)
+    )
+    last_file <- paths[length(paths)]
+    # Parse arguments to be passed to read_delim and set defaults
+    read_args <- list(...)
+    read_args$file <- replace_null(read_args$file, last_file)
+    read_args$delim <- replace_null(read_args$delim, ",")
+    read_args$progress <- replace_null(read_args$progress, FALSE)
```

Comment:
Personally I like to show always progress bars when reading files. The reason is quite simple. 
If the file is small, you don't notice the bar even and the processing time is still very small. If the file is big, you know why you are waiting for.